pkg/providers/instancetype/instancetype.go

```diff
@@ -157,7 +167,11 @@ func (p *DefaultProvider) List(ctx context.Context, kc *v1alpha1.KubeletConfigur
 	if p.cm.HasChanged("zones", allZones) {
 		log.FromContext(ctx).WithValues("zones", allZones.UnsortedList()).V(1).Info("discovered zones")
 	}
-
+	maxAvailableIPCount := lo.Max(
```

Comment:
If maxAvailableIPCount <= 0, we should directly return error, do I understand correct? @helen-frank